assets/js/components/income.js

```diff
@@ -132,24 +132,34 @@ export class LocalIncomeSection extends AbstractIncomeSection {
   }
 
   _initChartData() {
-    const items = this.sectionData.revenueBreakdown.values.filter((item) => item.amount_type === 'AUDA');
-    items.forEach((item) => item.color = localColor);
-    const yearGroups = _.groupBy(items, 'date');
-    this._year = _.max(_.keys(yearGroups));
-    this._chartData = yearGroups;
+    if (this.sectionData.revenueBreakdown) {
+      this._year = null;
+      this._chartData = null;
+    } else {
+      const items = this.sectionData.revenueBreakdown.values.filter((item) => item.amount_type === 'AUDA');
```

Comment:
"If this.sectionData.revenueBreakdown is truthy (presumably because it's set?), then initialise this._year and this._chartData. Otherwise use this.sectionData.revenueBreakdown to initialise this._year and this._chartData"

Doesn't seem right? In what circumstance is this.sectionData.revenueBreakdown false but contains useful data for this code to work?

Reply:
did you mean `if (!this.sectionData.revenueBreakdown)`?

Reply:
Made some changes to handle missing revenueBreakdown properly in the chart and dropdown